src/coap_client_libcoap.c

```diff
@@ -507,31 +507,6 @@ static void golioth_coap_delete(golioth_coap_request_msg_t *req, coap_session_t
     coap_send(session, req_pdu);
 }
 
-static void add_observation(golioth_coap_request_msg_t *req, struct golioth_client *client)
-{
-    // scan for available (not used) observation slot
-    golioth_coap_observe_info_t *obs_info = NULL;
-    bool found_slot = false;
-    for (int i = 0; i < CONFIG_GOLIOTH_MAX_NUM_OBSERVATIONS; i++)
-    {
-        obs_info = &client->observations[i];
-        if (!obs_info->in_use)
-        {
-            found_slot = true;
-            break;
-        }
-    }
-
-    if (!found_slot)
-    {
-        GLTH_LOGE(TAG, "Unable to observe path %s, no slots available", req->path);
-        return;
-    }
-
-    obs_info->in_use = true;
-    memcpy(&obs_info->req, req, sizeof(obs_info->req));
-}
-
 static void golioth_coap_observe(golioth_coap_request_msg_t *req,
```

Comment:
Can we make similar changes as in the Zephyr implementation to bubble up errors?